src/docs/reference/templates.md

```diff
@@ -107,3 +104,54 @@ You can see all of your templates on your [Account's Templates page](https://rai
  width={3100}
  quality={80}
 />
+
+### Sharing your Templates
+
+After you create your template, you may want to share your work with the public and let others clone your project.
+
+#### Deploy on Railway Button
+
+Upon template creation, you are provided with the Template URL where your template can be found and deployed.  
+
+To complement your template, we also provide a `Deploy on Railway` button which you can include in your README or embed it into a website.
+![https://railway.app/button.svg](https://railway.app/button.svg)
+The button is located at [https://railway.app/button.svg](https://railway.app/button.svg).
+
+For examples and more information about the button, head over to our [Deploy On Railway Button](/deploy/deploy-on-railway-button) page.
+
+#### Kickback program
+
+If your published template is deployed into other users' projects, you are immediately eligible for a 25% kickback, in the form of Railway credits, of the usage cost incurred by those users.  That means if a user deploys your template, and the usage of the services cost the user $100, you could receive $25 in Railway credits.
+
+Important things to note:
+- Your template must be published to the marketplace to be eligible for kickback.
+- For Hobby users with a $5 discount, only usage in excess of the discount is counted in the kickback.
+- Platform fees are not included in the kickback, but usage fees of the platform are included. E.g. if a user pays $20 in platform fees, then incurs $200 of usage from your template, you are eligible for a $50 kickback (25% of $200).
+- All service types and resource usage of those services (compute, volume, egress, etc) *do count* towards the kickback.
+
+Read more about the kickback program [here](https://railway.app/open-source-kickback).
+
+## Deployment
+
+Templates allow you to deploy a fully configured project that is automatically
+connected to infrastructure. Examples of templates are -
+
+- NextJS app with Prisma
+- Django app connected to Postgres
+- Elixir Phoenix webserver
+- Discord/Telegram bots
+
+You can find featured templates on our
+[dedicated templates page](https://railway.app/templates).
+
+### Updatable Templates
```

Comment:
Should we mention that updates only work for templates with GH repos and not Docker images?